app/src/main/java/com/humara/nagar/ui/report/ReportFragment.kt

```diff
@@ -23,23 +43,216 @@ class ReportFragment : BaseFragment() {
+        )
+        pictureDialog.setItems(
+            pictureDialogItems
+        ) { _, which ->
+            when (which) {
+                0 -> choosePhotoFromGallery()
+                1 -> takePhotoFromCamera()
+            }
+        }
+        val dialog = pictureDialog.create()
+        dialog.show()
+    }
+
+    private fun choosePhotoFromGallery() {
+        val permissions = arrayOf(
+            Manifest.permission.READ_EXTERNAL_STORAGE
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else {
+            //Pick multiple pictures (limit max 5)
         }
-        return root
+    }
+
+    private fun takePhotoFromCamera() {
+        val permissions = arrayOf(
+            Manifest.permission.CAMERA
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else {
+            //Take picture
+        }
+    }
+
+
+    @Deprecated("Deprecated in Java")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        var allPermissionsGranted = true
+        val deniedPermissions = mutableListOf<String>()
+        for (i in requestedPermissions.indices) {
+            if (grantResults[i] != PackageManager.PERMISSION_GRANTED) {
+                Logger.debugLog("NOT GRANTED ${requestedPermissions[i]}")
+                allPermissionsGranted = false
+                deniedPermissions.add(requestedPermissions[i])
+                break
+            } else {
+                Logger.debugLog("GRANTED ${requestedPermissions[i]}")
+                break
+            }
+        }
+        if (allPermissionsGranted) {
+            // All permissions were granted. You can proceed with the action that requires these permissions.
+            if (permissions.contains(Manifest.permission.ACCESS_FINE_LOCATION) &&
+                    permissions.contains(Manifest.permission.ACCESS_COARSE_LOCATION)) {
+                getAddress()
+            }
+        } else {
+            // One or more permissions were denied. You should show a message to the user explaining why you need these permissions.
+            Logger.debugLog("Denied permission: $deniedPermissions")
+            handlePermissionDenied(deniedPermissions.toTypedArray())
+        }
+
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    }
+
+    fun handlePermissionDenied(permissions: Array<String>) {
+        val showPermissionRationalePopup: Boolean = permissions.isNotEmpty()
```

Comment:
`checkUserRequestedDontAskAgain` function in PermissionUtil class is for the same

Reply:
Removed: `checkUserRequestedDontAskAgain` as `handlePermissionDenied` was added in the PermissionsUtils.kt